lib/event.rb

```diff
@@ -0,0 +1,30 @@
+class Event
+
+  attr_reader :name, :crafts, :attendee_names
+
+  def initialize(name, craft, names)
+    @name = name
+    @crafts = [].push(craft).flatten
+    @attendee_names = [].push(names).flatten
+  end
+
+  def craft_with_most_supplies
+    most = @crafts.max {|a, b| a.supplies_required.length <=> b.supplies_required.length}
+    most.name
+  end
+
+  def supply_list
+    supply_array = @crafts.flat_map {|craft| craft.supplies_required}
+    item_names = []
+    supply_array.each do |hash|
+      item_names << hash.keys
+    end
```

Comment:
```suggestion
    supply_array = @crafts.flat_map {|craft| craft.supplies_required.keys}
```

I think if you grab the keys in the initial flat_map it could simplify this method a bit